bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -75,11 +75,20 @@ type KubeadmConfigSpec struct {
 	// +optional
 	Mounts []MountPoints `json:"mounts,omitempty"`
 
-	// preKubeadmCommands specifies extra commands to run before kubeadm runs
+	// bootCommands specifies extra commands to run very early in the boot process via the cloud-init bootcmd
+	// module. This is typically run in the cloud-init.service systemd unit. This has no effect in Ignition.
```

Comment:
I just learned that bootCommands run on every boot (`always`), I think that's worth to mention.

Reply:
Sure, I wrote what they say in the [docs](https://cloudinit.readthedocs.io/en/latest/reference/examples.html#run-commands-on-first-boot)